### PR TITLE
PAL-976 Use shared redis/kafka

### PR DESCRIPTION
**Pull request by kg981167224**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -13,8 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-@Library('jenkinsfile-lib')_
+@Library('jenkinsfile-lib@feature/PAL-976-update-smoke-tests-shared-redis-kafka')_
```
Comment (kg981167224) [inline Jenkinsfile]: Need to remove this before merge, after the groovy scripts have been merged

File: NOTICES.md [hunk 2]
```diff
@@ -40,9 +40,9 @@ List of third-party dependencies grouped by their license type
 * Spring Cloud Config Client ([org.springframework.cloud:spring-cloud-config-client:2.2.2.RELEASE](https://spring.io))
 * Spring Cloud Starter OpenFeign ([org.springframework.cloud:spring-cloud-starter-openfeign:2.2.2.RELEASE](https://projects.spring.io/spring-cloud))
 * Spring Kafka Support ([org.springframework.kafka:spring-kafka:2.5.2.RELEASE](https://github.com/spring-projects/spring-kafka))
-* common ([uk.gov.gchq.palisade:common:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-common))
-* hadoop-reader ([uk.gov.gchq.palisade:hadoop-reader:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-reader))
-* hadoop-resource ([uk.gov.gchq.palisade:hadoop-resource:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-resource))
+* common ([uk.gov.gchq.palisade:common:0.5.0-20210127.140835-23](https://github.com/gchq/Palisade-common))
+* hadoop-reader ([uk.gov.gchq.palisade:hadoop-reader:0.5.0-20210126.160642-27](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-reader))
+* hadoop-resource ([uk.gov.gchq.palisade:hadoop-resource:0.5.0-20210126.160603-27](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-resource))
```
Comment (pd104923) [inline NOTICES.md]: These need to be reverted back to `SNAPSHOT`
```suggestion
* common ([uk.gov.gchq.palisade:common:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-common))
* hadoop-reader ([uk.gov.gchq.palisade:hadoop-reader:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-reader))
* hadoop-resource ([uk.gov.gchq.palisade:hadoop-resource:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/hadoop-resource))
```

Comment (developer01189998819991197253) [inline NOTICES.md]: revert these changes too!

File: README.md [hunk 3]
```diff
@@ -166,5 +166,4 @@ All instances of the palisade-service pod contain a script that will identify al
 ```
 kubectl exec -it palisade-service-7bb4d75d85-g8cgx -- bash /usr/share/palisade-service/log-level -t
 ```
-
```
Comment (developer01189998819991197253) [inline README.md]: line change intentional?

File: licenses/cddl_gplv2_ce.html [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 
 <!DOCTYPE html>
-<html lang="en">
+<html lang="en" >
```
Comment (developer01189998819991197253) [inline licenses/cddl_gplv2_ce.html]: revert license changes

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]
Helm changes approved, just some license changes

File: values.yaml [hunk 5]
```diff
@@ -35,6 +35,9 @@ global:
   # hostname: the external hostname of the ingress, this may be the cluster master or public load-balancer
   hostname: localhost
 
+  # If specified is the service name to use typically in a piepline this will be the
```
Comment (pd104923) [inline values.yaml]: I'm not sure this comment makes sense. I don't know what you're trying to say here.

**Review by pd104923 — CHANGES_REQUESTED** [review]
Can the license file be restored back to what was in develop?
NOTES.md needs to be reverted back to `SNAPSHOT` as well
Comment in values.yaml does not make much sense when reading it.

File: NOTICES.md (hunk 2 above)
Comment (kg981167224) [reply NOTICES.md]: Done

**Review by developer01189998819991197253 — DISMISSED** [review]

**Review by pd104923 — DISMISSED** [review]

File: charts/attribute-masking-service/templates/configmap-akka-redis.yaml [hunk 6]
```diff
@@ -28,11 +28,14 @@ data:
...
           input-topic.partitions: {{ .Values.replicaCount }}
+          {{ if .Values.global.topicPrefix }}input-topic.name: {{ .Values.global.topicPrefix }}rule {{ end }}
         producer.topics:
           output-topic.partitions: {{ .Values.replicaCount }}
+          {{ if .Values.global.topicPrefix }}output-topic.name: {{ .Values.global.topicPrefix }}masked-resource {{ end }}
           error-topic.partitions: {{ .Values.replicaCount }}
+          {{ if .Values.global.topicPrefix }}error-topic.name: {{ .Values.global.topicPrefix }}error {{ end }}
       discovery.config.services.kafka.endpoints:
-      - host: {{ include "palisade.kafka.fullname" . }}-headless
+      - host: {{ include "palisade.kafka.fullname" . }}-headless{{ .Values.global.serviceName }}
```
Comment (dev930018) [inline charts/attribute-masking-service/templates/configmap-akka-redis.yaml]: Would prefer a different mechanism, maybe consider if we can not use the headless svc

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: charts/attribute-masking-service/templates/configmap-akka-redis.yaml (hunk 6 above)
Comment (dev-958) [reply charts/attribute-masking-service/templates/configmap-akka-redis.yaml]: this doesn't seem right, let's have a talk about it when you're ready

**Review by dev-958 — CHANGES_REQUESTED** [review]

**Review by pd104923 — DISMISSED** [review]

File: templates/kafka-proxy-svc.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  name: palisade-kafka-proxy
+  namespace: {{ include "palisade.namespace" . }}
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  ports:
+  - name: palisade-kafka
+    port: 9092
+
+  type: ExternalName
```
Comment (dev930018) [inline templates/kafka-proxy-svc.yaml]:
```suggestion
    port: 9092
  type: ExternalName
```

File: templates/redis-cluster-proxy-svc.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  name: palisade-redis-proxy-cluster
+  namespace: {{ include "palisade.namespace" . }}
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  ports:
+  - name: palisade-redis-cluster
+    port: 16379
+
+  type: ExternalName
```
Comment (dev930018) [inline templates/redis-cluster-proxy-svc.yaml]:
```suggestion
    port: 16379
  type: ExternalName
```

File: templates/redis-proxy-svc.yaml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  name: palisade-redis-proxy-master
+  namespace: {{ include "palisade.namespace" . }}
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  ports:
+  - name: palisade-redis
+    port: 6379
+
+  type: ExternalName
```
Comment (dev930018) [inline templates/redis-proxy-svc.yaml]:
```suggestion
    port: 6379
  type: ExternalName
```

File: templates/zookeeper-proxy-svc.yaml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  name: palisade-zookeeper-proxy
+  namespace: {{ include "palisade.namespace" . }}
+  annotations:
+    service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+spec:
+  ports:
+  - name: palisade-zookeeper
+    port: 2181
+
+  type: ExternalName
```
Comment (dev930018) [inline templates/zookeeper-proxy-svc.yaml]:
```suggestion
    port: 2181
  type: ExternalName
```

File: Jenkinsfile (hunk 1 above)
Comment (dev930018) [inline Jenkinsfile]:
```suggestion
@Library('jenkinsfile-lib')_
```

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (dev930018) [reply Jenkinsfile]: Whenever ready, of course

**Review by dev-958 — DISMISSED** [review]

**Review by developer01189998819991197253 — DISMISSED** [review]

**Review by dev930018 — APPROVED** [review]
